canonicalwebteam/flask_base/app.py

```diff
@@ -0,0 +1,29 @@
+import flask
+import talisker.flask
+
+from werkzeug.contrib.fixers import ProxyFix
+from werkzeug.debug import DebuggedApplication
+
+from canonicalwebteam.flask_base.converters import RegexConverter
+
+
+class FlaskBase(object):
+    def __init__(self, app=None, service=None):
```

Comment:
I know this `app=None` interface is the [recommended way](http://flask.pocoo.org/docs/1.0/extensiondev/#the-extension-code) to do a Flask application extension. This is so that you can either create a new application or extend an existing one.

But could I suggest in this case that we change this interface to better communicate our intention? This is intended to be an opinion app, to be used as the basis for everything else. Therefore I don't think we're ever intending to extend an existing application with it. So couldn't we remove the `init_app` method, put everything in `__init__`, and not accept an `app` argument? This makes it clear that we're expecting you to do:

``` python3
from canonicalwebteam.flask_base import FlaskBase

app = FlaskBase()

# ... some other stuff
```

And explicitly not:

``` python3
from flask import Flask
from canonicalwebteam.flask_base import FlaskBase

app = Flask()

# .. some other stuff

app = FlaskBase.init_app(app)
```

I think being explicit about this means we don't need to make sure we test using it both ways, make sure it's not going to clobber existing application settings etc. It will make our lives simpler and likely avoid bugs in the latter case.